tests/nativeapp/codegen/snowpark/test_extension_function_utils.py

```diff
@@ -1,94 +1,195 @@
-            file_path=dest_path,
-            suffix_str_to_rm=".py",
-            deploy_root=Path(local_path, "output", "deploy"),
-        )
-        assert actual == expected
+# --------------------------------------------------------
+# ------------- get_object_type_as_text ------------------
+# --------------------------------------------------------
 
 
 @pytest.mark.parametrize(
     ("input_param, expected"),
     [
         ("TABLE_FUNCTION", "TABLE FUNCTION"),
         ("FUNCTION", "FUNCTION"),
+        ("AGGREGATE-FUNCTION", "AGGREGATE-FUNCTION"),
     ],
 )
 def test_get_object_type_as_text(input_param, expected):
-    actual = ef_utils._get_object_type_as_text(input_param)  # noqa: SLF001
+    actual = ef_utils.get_object_type_as_text(input_param)
     assert actual == expected
 
 
-@mock.patch(
-    "snowflake.cli.plugins.nativeapp.codegen.snowpark.extension_function_utils._get_handler_path_without_suffix"
-)
-@pytest.mark.parametrize(
-    ("dest_file, func, expected, helper_ret"),
-    [
-        ("a/b/c/d.py", "DUMMY_FUNC", "a.b.c.d.DUMMY_FUNC", "a.b.c.d"),
-        ("a/b/c/d.py", ("FUNC1", "FUNC2"), None, "a.b.c.d"),
-    ],
-)
-def test_get_handler(mock_helper, dest_file, func, expected, helper_ret):
-    mock_helper.return_value = helper_ret
-    actual = ef_utils._get_handler(  # noqa: SLF001
-        Path(dest_file), func, deploy_root=Path("some/path")
-    )
-    assert actual == expected
+# --------------------------------------------------------
+# --------- sanitize_extension_function_data -------------
+# --------------------------------------------------------
 
 
-@pytest.mark.parametrize(
-    ("object_name, schema, handler, expected"),
-    [
-        ("SNOWPARK_TEMP_DUMMY", "core", "DUMMY_HANDLER", "core.DUMMY_HANDLER"),
-        ("DUMMY_HANDLER", "core", "DUMMY_HANDLER", "core.DUMMY_HANDLER"),
-    ],
-)
-def test_get_object_name_for_udf_sp(object_name, schema, handler, expected):
-    actual = ef_utils._get_schema_and_name_for_extension_function(  # noqa: SLF001
-        object_name, schema, handler
+def test_sanitize_extension_function_data_required_keys(snapshot):
+
+    ex_fn = {}
+    some_path = Path("some/path")
+
+    # Test for absence or malformed object_type, object_name and return_sql
+    with pytest.raises(ef_utils.MalformedExtensionFunctionError) as err:
+        ef_utils.sanitize_extension_function_data(ex_fn=ex_fn, py_file=some_path)
+    assert err.value.message.__contains__("object_type")
+
+    ex_fn["object_type"] = ["function"]
+    with pytest.raises(ef_utils.MalformedExtensionFunctionError) as err:
+        ef_utils.sanitize_extension_function_data(ex_fn=ex_fn, py_file=some_path)
+    assert err.value.message.__contains__("object_type")
+
+    ex_fn["object_type"] = "function"
+    with pytest.raises(ef_utils.MalformedExtensionFunctionError) as err:
+        ef_utils.sanitize_extension_function_data(ex_fn=ex_fn, py_file=some_path)
+    assert err.value.message.__contains__("object_name")
```

Comment:
took me a minute to figure out the pattern here. You're basically building up to a a correct dictionary, so you keep asserting that the next thing to fix is flagged. Maybe add that as a comment to help the next person reading this?